test/integration/shared/host_orchestration_stubs.rb

```diff
@@ -0,0 +1,19 @@
+module HostOrchestrationStubs
+  extend ActiveSupport::Concern
+
+  def click_on_submit
+    click_button('Submit')
+    wait_for_orchestration_requests # host-progress bar should disappear
+    wait_for_ajax # wait for runtime/nics/etc.. ajax requests in Host#show
+    assert page.has_link?('Edit')
+  end
+
+  def wait_for_orchestration_requests
+    Timeout.timeout(Capybara.default_max_wait_time * 2) do
+      # Notice host-progress won't be visible without a multi-threaded server
+      loop until has_no_selector?("#host-progress", :visible => false, :wait => 0.1)
+    end
+  rescue Capybara::ElementNotFound
```

Comment:
I ran a few jobs on Jenkins to test this, one failed with this:

<pre>
Capybara::Poltergeist::ObsoleteNode: The element you are trying to interact with is either not part of the DOM, or is not currently visible on the page (perhaps display: none is set). It's possible the element has been replaced by another element and you meant to interact with the new element. If so you need to do a new 'find' in order to get a reference to the new element.
    test/integration/shared/host_orchestration_stubs.rb:14:in `block in wait_for_orchestration_requests'
    test/integration/shared/host_orchestration_stubs.rb:12:in `wait_for_orchestration_requests'
    test/integration/shared/host_orchestration_stubs.rb:6:in `click_on_submit'
    test/integration/host_js_test.rb:168:in `block (2 levels) in <class:HostJSTest>' (Capybara::Poltergeist::ObsoleteNode)
/usr/local/rvm/gems/ruby-2.1.5@test_develop_pr_core-0/gems/capybara-2.6.2/lib/capybara/node/matchers.rb:53    
</pre>

(http://ci.theforeman.org/job/test_develop_pr_core/8566/database=postgresql,label=fast,ruby=2.1/)